### ISSUE TITLE
vo:gpu-next shader Anime4k_restore would cause the aliasing

### ISSUE THREAD

**Issue by hooke007** [mpv-player/mpv#9505]
### Important Information

Provide following Information:
- mpv version
4ca6c2ff342cf40c8ff7fc5271df357465098942
- Windows Version
win11 release
- Source of the mpv binary
shinchirol's toolchain
- If known which version of mpv introduced the problem
17df53519f6359881462da0251e85f99fefd884d is the last safe commit
- Possible screenshot or video of visual glitches
smooth 17df53519f6359881462da0251e85f99fefd884d
![Snipaste_2021-11-23_01-45-41](https://user-images.githubusercontent.com/41094733/142910756-f08e2251-3d34-4db3-a177-3efdf253208f.jpg)
aliasing 4ca6c2ff342cf40c8ff7fc5271df357465098942
![Snipaste_2021-11-23_01-38-31](https://user-images.githubusercontent.com/41094733/142910855-d2e525cf-276c-468a-9f0a-3776392ce283.jpg)

### Reproduction steps

1. start mpv by vo=gpu-next
2. open the video
3. apply shader Anime4k restore

BTW, vo_gpu does not have the issue.

### Expected behavior
No aliasing

### Actual behavior
see previous pictures

### Log file
[4ca6c2ff342cf40c8ff7fc5271df357465098942.log](https://github.com/mpv-player/mpv/files/7583294/4ca6c2ff342cf40c8ff7fc5271df357465098942.log)

[17df53519f6359881462da0251e85f99fefd884d.log](https://github.com/mpv-player/mpv/files/7583299/17df53519f6359881462da0251e85f99fefd884d.log)

### Sample files

https://github.com/bloc97/Anime4K/blob/master/glsl/Restore/Anime4K_Restore_CNN_M.glsl

**Comment by haasn** [comment]
Can't reproduce. Can you provide the exact command line (with `--no-config`) and/or a sample file?

Also, you seem to be using two different libplacebo versions in each. It's entirely possible the issue is due to a change somewhere in libplacebo, rather than mpv.

**Comment by hooke007** [comment]
`./mpv --no-config --vo=gpu-next --log-file=gpu-next.log 1280x720.mp4`
[gpu-next.log](https://github.com/mpv-player/mpv/files/7584044/gpu-next.log)

https://user-images.githubusercontent.com/41094733/142929432-dd0d3d44-fff7-42fd-8eff-c2cccfcfae5a.mp4


Maybe it was caused by libplacebo's commits, but I cannot check for I cannot specify the version of dependency.

**Comment by hooke007** [comment]
BTW, I cannot reproduce it 100% while playing, it could happen sometimes.

**Comment by hooke007** [comment]
Now I am aware of the condition, it only happens at paused stat.

**Comment by haasn** [comment]
Is this still an issue?

**Comment by hooke007** [comment]
Yes. (Last month's test) https://github.com/bloc97/Anime4K/issues/144#issuecomment-1234859115
If fixed I would edit.

**Comment by hooke007** [comment]
It's quite odd. I found another "stable method"(enable console) to reproduce.
Now it don't care about orginal size. If scaling and paused, the artifacts would appear.

https://user-images.githubusercontent.com/41094733/192657757-6fe0d092-c4c3-497e-9cb1-d3df9cf28ff5.mp4

**Comment by igv** [comment]
Actually the Restore shader always causes aliasing for low res videos (the lower the more aliasing), because it's for 1920x1080 resolution only. Maybe somehow under vulkan that 384x288 resolution is used for storing temporary textures? I can't reproduce this, it's identical whether gpu or gpu-next and always with more aliasing than without the shader (video resolution is 960x540), but I haven't tried a video with subtitles. Also maybe it's nvidia only issue again...

Btw, why under gpu-next, png screenshots are much larger than under gpu?

**Comment by yeezylife** [comment]
> Btw, why under gpu-next, png screenshots are much larger than under gpu?

I noticed that too. I guess that's because `screenshot-png-compression=` doesn't work properly with gpu-next.

I just use lossless webp for screenshots now...

**Comment by haasn** [comment]
> I noticed that too. I guess that's because `screenshot-png-compression=` doesn't work properly with gpu-next.

??? Why wouldn't it? The screenshot writing code is the exact same. Do you have examples of such screenshots (taken with identical settings) under gpu and gpu-next respectively?

My guess would be that it's somehow related to dithering.

**Comment by haasn** [comment]
> > I noticed that too. I guess that's because `screenshot-png-compression=` doesn't work properly with gpu-next.
> 
> ??? Why wouldn't it? The screenshot writing code is the exact same. Do you have examples of such screenshots (taken with identical settings) under gpu and gpu-next respectively?
> 
> My guess would be that it's somehow related to dithering.

Nvm, I can reproduce it on my end, and the reason is that `--screenshot-high-bit-depth`, which defaults to `yes`, apparently works for `gpu-next` but not for `gpu`. I get 16-bit PNG screenshots out of `gpu-next` out of the box. If I disable that option I get the same output on both.

**Comment by hooke007** [comment]
There is no subtitle in the video (The first time I posted).

**Comment by igv** [comment]
Add to the description that you need to remove `//!SAVE MAIN` from the shader now.

Looks suspicious that it only happens when a video is paused, zooming-in also doesn't work in that case and that this bug appeared after a fix to panning...

**Comment by haasn** [comment]
> Add to the description that you need to remove //!SAVE MAIN from the shader now.

Isn't this fixed by https://code.videolan.org/videolan/libplacebo/-/commit/33ded0c18b204d546b91d6d1e36e9b0b6feb23d5 already?

**Comment by igv** [comment]
Idk, I'm still on v4.208.

**Comment by igv** [comment]
> (Re)creating 1952x247x0 texture with format r8
> (Re)creating 2045x247x0 texture with format r8
What's this? This is right before `Set property: pause -> 1`

**Comment by haasn** [comment]
> > (Re)creating 1952x247x0 texture with format r8
> > (Re)creating 2045x247x0 texture with format r8
> 
> What's this? This is right before `Set property: pause -> 1`

Possibly something used for internal redraw cache (on pause), or OSD texture

**Comment by igv** [comment]
> Possibly something used for internal redraw cache (on pause)

Not only before `Set property: pause -> 1`. There is also `(Re)creating 1616x174x0 texture with format r8` right after `Registering hook pass: Anime4K`.

> or OSD texture

mpv window size is 1920x1080

**Comment by hooke007** [comment]
Almost forgot it. Cannot reproduce it now.
